src/index.ts

```diff
@@ -138,6 +140,66 @@ export const createInstance = (config: Config): Instance => {
 
       return tx;
     },
+
+    buyNow: async (
+      signer: ethers.Signer,
+      params: BuyNowParams
+    ): Promise<ethers.ContractTransaction> => {
+      const zAuction = await getZAuctionContract(
+        signer,
+        config.zAuctionAddress
+      );
+
+      const nftContract = await getERC721Contract(
+        config.web3Provider,
+        config.tokenContract
+      );
+
+      const isApproved = await actions.isZAuctionApprovedNftTransfer(
+        await signer.getAddress(),
+        config.zAuctionAddress,
+        nftContract
+      );
+
+      if (!isApproved) throw Error("User is not approved for NFT Transfer");
+
+      const tx = await zAuction.buyNow(
+        params.auctionId,
+        params.amount,
+        params.tokenId,
+        params.startBlock,
+        params.expireBlock
+      );
+
+      return tx;
+    },
+
+    setBuyNow: async (
+      signer: ethers.Signer,
+      params: SetBuyNowParams
+    ): Promise<ethers.ContractTransaction> => {
+      const zAuction = await getZAuctionContract(
+        signer,
+        config.zAuctionAddress
+      );
+
+      if (signer.getAddress() !== zAuction.owner()) throw Error("Cannot set the price of a domain that is not yours")
```

Comment:
`zAuction.owner()` returns the owner of the `zAuction` smart contract, not the owner of the domain `params.tokenid`.

You will need to access the Registrar to get the owner of a token.

Reply:
Registrar === nftContract

```
const nftContract = await getERC721Contract(
        config.web3Provider,
        config.tokenContract
      );
nftContract.ownerOf(tokenid)
```